lib/sms/sms_submit.c

```diff
@@ -88,6 +104,93 @@ static int sms_submit_encode_number(
 	return 0;
 }
 
+/**
+ * @brief Encode SMS Service Center Address (SCA) field.
+ *
+ * @details SCA field is used in SMS-SUBMIT message as specified in 3GPP TS 23.040 chapter
+ *
+ * @param[out] sca_field Buffer for SCA field.
+ *                       Must be at least 23 bytes long to include the following information:
+ *                       - Address/number: 20
+ *                       - Address length: 1
+ *                       - Type of address: 1
+ *                       - NUL termination: 1
+ */
+static void sms_submit_encode_sca(char *sca_field)
+{
+	int ret;
+
+	uint8_t encoded_sca[SMS_MAX_ADDRESS_LEN_CHARS + 1];
+	uint8_t encoded_sca_size;
+	uint8_t encoded_sca_size_octets = 0;
+	/* Make buffer big enough and have few characters extra. Maximum required should be 41:
+	 * - "+CSCA: ": 7
+	 * - Address/number: 20
+	 * - plus sign: 1
+	 * - double quotes for address: 2
+	 * - comma: 1
+	 * - Type of address: 3
+	 * - \r\nOK\r\n: 6
+	 * - NUL termination: 1
+	 */
+	char csca_resp[48] = { 0 };
+	/* +1 for NUL termination and another +1 for potential '+' sign */
+	char sca[SMS_MAX_ADDRESS_LEN_CHARS + 2] = { 0 };
+	int tosca = 0x91;
```

Comment:
This value is in decimal in all other places, it could be clearer to use `145` here. Maybe also add a comment where this value comes from.

Reply:
I'm adding a comment and changed it to decimal format. However, it's in decimal format in `AT+CSCA` AT command, but in hexadecimal format in the encoded number in `AT+CMGS` (see `sms_submit_encode`)